File: ClientApp/src/app/features/reservations/classes/resources/reservation-write-resource.ts
import { Guid } from "guid-typescript"

export class ReservationWriteResource {

    constructor(
        public reservationId: Guid,
        public date: string)
    // public destinationId: number,
    // public customerId: number,
    // public pickupPointId: number,
    // public portId: number,
    // public driverId: number,
    // public shipId: number
    // public ticketNo: string,
    // public email: string,
    // public phones: string,
    // public adults: number,
    // public kids: number,
    // public free: number,
    // public remarks: string,
    // public guid: string,
    // public userId: string,
    // public passengers: PassengerWriteResource[])
    { }
}

File: API/Features/Reservations/Resources/Form/ReservationWriteResource.cs
using System;
using System.Collections.Generic;

namespace API.Features.Reservations {

    public class ReservationWriteResource {

        public Guid? ReservationId { get; set; }
        public int CustomerId { get; set; }
        public int DestinationId { get; set; }
        public int PickupPointId { get; set; }
        public int PortId { get; set; }
        public int? DriverId { get; set; }
        public int? ShipId { get; set; }
        public string Date { get; set; }
        public string RefNo { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public string Remarks { get; set; }
        public string UserId { get; set; }

        public List<PassengerWriteResource> Passengers { get; set; }

    }

}